raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -275,296 +278,299 @@ contract TokenNetwork is Utils {
-    // called multiple times by anyone, as long as they provide signatures from both participants.
+    // called multiple times by anyone.
+    /// @param balance_hash Hash of (transferred_amount, locked_amount, locksroot).
+    /// @param additional_hash Computed from the message. Used for message authentication.
     /// @param channel_identifier The channel identifier - mapping key used for `channels`.
     /// @param nonce Strictly monotonic value used to order transfers.
-    /// @param transferred_amount Total amount of tokens transferred by the channel partner
-    /// to the channel participant who calls the function.
-    /// @param locksroot Root of the partner's merkle tree of all pending lock lockhashes.
-    /// @param additional_hash Computed from the message. Used for message authentication.
     /// @param closing_signature Closing participant's signature of the balance proof data.
     /// @param non_closing_signature Non-closing participant signature of the balance proof data.
-    function updateTransferDelegate(
+    function updateNonClosingBalanceProof(
         uint256 channel_identifier,
-        uint64 nonce,
-        uint256 transferred_amount,
-        bytes32 locksroot,
+        bytes32 balance_hash,
+        uint256 nonce,
         bytes32 additional_hash,
         bytes closing_signature,
         bytes non_closing_signature)
         external
     {
-        // We also need the signature from the non-closing participant on behalf of which the 3rd
-        // party makes the transaction. This signature will be provided to the 3rd party
-        //  together with the balance proof.
+        Channel storage channel = channels[channel_identifier];
+
+        // Channel must be closed
+        require(channel.state == 2);
+
+        // Channel must be in the settlement window
+        require(channel.settle_block_number >= block.number);
+
+        // We need the signature from the non-closing participant to allow anyone
+        // to make this transaction. E.g. a monitoring service.
         address non_closing_participant = recoverAddressFromBalanceProof(
             channel_identifier,
+            balance_hash,
             nonce,
-            transferred_amount,
-            locksroot,
             additional_hash,
             non_closing_signature
         );
 
-        // Make sure the second signature is from a channel participant
-        require(channels[channel_identifier].participants[non_closing_participant].initialized);
+        Participant storage non_closing_participant_state = channel.participants[
+            non_closing_participant
+        ];
 
-        // Make sure the second signature is from the non-closing participant
-        require(closing_requests[channel_identifier].closing_participant != non_closing_participant);
+        // Make sure the signature is from a channel participant
+        require(non_closing_participant_state.initialized);
```

Comment:
IMO this should not take priority over clarity if the input is **invalid**, only in cases where the input is valid but a race happened (e.g. two monitoring services called update at the same time, only one of the transactions will work)

Reply:
No need to change it in this PR but I agree with Augusto. Clarity should be what we aim for in smart contract code rather than providing tiny bit more gas back with `REVERT`

Reply:
keeping track of this here https://github.com/raiden-network/raiden-contracts/issues/47